lib/matplotlib/tests/test_axes_clear_shared_ticks.py

```
@@ -0,0 +1,76 @@
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+
+
+def test_clear_respects_tick_side_defaults():
+    with mpl.rc_context({
+        "xtick.top": False,
+        "ytick.right": False,
+        "xtick.minor.top": False,
+        "ytick.minor.right": False,
+        "xtick.minor.visible": True,
+        "ytick.minor.visible": True,
+    }):
+        fig, ax = plt.subplots()
+        try:
+            ax.plot([0, 1], [0, 1])
+            ax.clear()
+            ax.plot([0, 1], [0, 1])
+            ax.minorticks_on()
+            fig.canvas.draw()
+
+            assert all(
+                not tick.tick2line.get_visible()
+                for tick in ax.xaxis.get_major_ticks()
+            )
+            assert all(
+                not tick.tick2line.get_visible()
+                for tick in ax.yaxis.get_major_ticks()
+            )
+            x_minor = ax.xaxis.get_minor_ticks()
+            if x_minor:
```

[minor] Could we assert that the minor tick lists are actually populated before checking their visibility? As written, the `if x_minor:` guard skips the assertions whenever no minor ticks are present, so the test could produce a false positive. An explicit `assert x_minor` (and the analogous check for y) would keep the test honest and still make the visibility assertion meaningful.

---

Added explicit assertions that the minor tick lists are populated before checking visibility so the regression test can’t silently pass with empty lists.